### PR TITLE
Amazon Q Code Transform: Improve download error handling ux

### PR DESCRIPTION
**Pull request by tincheng**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description

- Show errors in Code Transform chat panel when encountering download errors.
- Remove redundant download error notification. 

image

image

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt [hunk 1]
```diff
@@ -457,6 +465,11 @@ class CodeTransformChatController(
         }
     }
 
+    private suspend fun handleDownloadFailed(failureReason: DownloadFailureReason) {
+        codeTransformChatHelper.addNewMessage(buildDownloadFailureChatContent(failureReason))
+        codeTransformChatHelper.addNewMessage(buildStartNewTransformFollowup())
```
Comment (damntrecky) [inline plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt:470]: Not a blocker: How many failure functions call `buildStartNewTransformFollowup`? 6-7? 

Its out of scope for this PR, but would be a nice to have this abstracted so  any failure state that calls `buildStartNewTransformFollowup` goes through a single function.

**Review by damntrecky — APPROVED** [review]

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt [hunk 2]
```diff
@@ -62,6 +64,10 @@ const val MAX_ZIP_SIZE = 1000000000 // 1GB
 const val HIL_1P_UPGRADE_CAPABILITY = "HIL_1pDependency_VersionUpgrade"
 const val EXPLAINABILITY_V1 = "EXPLAINABILITY_V1"
 
+// constants for handling SDKClientException
+const val CONNECTION_REFUSED_ERROR: String = "Connection refused"
```
Comment (manodnyab) [inline plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt:68]: Are these user facing strings? Should they be localized?

File: .changes/next-release/feature-751c6024-4e83-46c2-bbd3-d6222d39afe8.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "feature",
+  "description" : "Amazon Q Code Transform: Improve download failure ux"
```
Comment (manodnyab) [inline .changes/next-release/feature-751c6024-4e83-46c2-bbd3-d6222d39afe8.json:3]: Is it possible to add more description/elaborate on what ux is changing here?

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt (hunk 2 above)
Comment (tincheng) [reply plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt:68]: These are not user facing strings - these are used for understanding specific errors from SDKClientException

File: .changes/next-release/feature-751c6024-4e83-46c2-bbd3-d6222d39afe8.json (hunk 3 above)
Comment (tincheng) [reply .changes/next-release/feature-751c6024-4e83-46c2-bbd3-d6222d39afe8.json]: done!

**Review by manodnyab — APPROVED** [review]

**Pull request merged**